models/cable/R/write.config.CABLE.R

```diff
@@ -0,0 +1,125 @@
+    hostteardown <- paste(hostteardown, sep = "\n", paste(settings$host$postrun, collapse = "\n"))
+  }
+  
+  # create job.sh
+  jobsh <- gsub("@HOST_SETUP@", hostsetup, jobsh)
+  jobsh <- gsub("@HOST_TEARDOWN@", hostteardown, jobsh)
+  
+  jobsh <- gsub("@SITE_LAT@", settings$run$site$lat, jobsh)
+  jobsh <- gsub("@SITE_LON@", settings$run$site$lon, jobsh)
+  jobsh <- gsub("@SITE_MET@", settings$run$site$met, jobsh)
+  
+  jobsh <- gsub("@START_DATE@", settings$run$start.date, jobsh)
+  jobsh <- gsub("@END_DATE@", settings$run$end.date, jobsh)
+  
+  jobsh <- gsub("@OUTDIR@", outdir, jobsh)
+  jobsh <- gsub("@RUNDIR@", rundir, jobsh)
+  
+  jobsh <- gsub("@BINARY@", settings$model$binary, jobsh)
+  
+  writeLines(jobsh, con = file.path(settings$rundir, run.id, "job.sh"))
+  Sys.chmod(file.path(settings$rundir, run.id, "job.sh"))
+  
+  #-----------------------------------------------------------------------
+  ### Edit a templated config file for runs
+  if (!is.null(settings$model$config) && file.exists(settings$model$config)) {
+    config.text <- readLines(con = settings$model$config, n = -1)
+  } else {
+    filename <- system.file(settings$model$config, package = "PEcAn.CABLE")
+    if (filename == "") {
+      if (!is.null(settings$model$revision)) {
+        filename <- system.file(paste0("config.", settings$model$revision), package = "PEcAn.CABLE")
+      } else {
+        model <- db.query(paste("SELECT * FROM models WHERE id =", settings$model$id), params = settings$database$bety)
+        filename <- system.file(paste0("config.r", model$revision), package = "PEcAn.CABLE")
+      }
+    }
+    if (filename == "") {
+      logger.severe("Could not find config template")
+    }
+    logger.info("Using", filename, "as template")
+    config.text <- readLines(con = filename, n = -1)
+  }
+  
+  config.text <- gsub("@SITE_LAT@", settings$run$site$lat, config.text)
+  config.text <- gsub("@SITE_LON@", settings$run$site$lon, config.text)
+  config.text <- gsub("@SITE_MET@", settings$run$inputs$met$path, config.text)
+  config.text <- gsub("@MET_START@", settings$run$site$met.start, config.text)
+  config.text <- gsub("@MET_END@", settings$run$site$met.end, config.text)
+  config.text <- gsub("@START_MONTH@", format(startdate, "%m"), config.text)
+  config.text <- gsub("@START_DAY@", format(startdate, "%d"), config.text)
+  config.text <- gsub("@START_YEAR@", format(startdate, "%Y"), config.text)
+  config.text <- gsub("@END_MONTH@", format(enddate, "%m"), config.text)
+  config.text <- gsub("@END_DAY@", format(enddate, "%d"), config.text)
+  config.text <- gsub("@END_YEAR@", format(enddate, "%Y"), config.text)
+  config.text <- gsub("@OUTDIR@", settings$host$outdir, config.text)
+  config.text <- gsub("@ENSNAME@", run.id, config.text)
+  config.text <- gsub("@OUTFILE@", paste0("out", run.id), config.text)
+  
+  #-----------------------------------------------------------------------
+  config.file.name <- paste0("CONFIG.", run.id, ".txt")
```

Comment:
Likewise, the file you write out needs to be named what CABLE expects it config file to be named (the doc suggests this is called cable.nml) https://trac.nci.org.au/trac/cable/wiki/CableUserGuide/Configuration